### PR TITLE
Windows registration command for `Flannel` CNI

### LINKED ISSUE #14828: RKE2 Custom Cluster UI does not show Windows registration command when flannel CNI is selected
**Setup**
- Rancher version: 2.11, 2.12
- Rancher UI Extensions: n/a
- Browser type & version: n/a

**Describe the bug**
RKE2 Custom Cluster UI does not show Windows registration command when flannel CNI is selected

**To Reproduce**
* Create new RKE2 custom cluster
* Select flannel CNI
* Create cluster

**Result**
Windows node registration command is not shown

**Expected Result**
Windows node registration command should be shown. Flannel has been supported for Windows nodes since the March 2024 release cycle (v1.29.2+rke2r1)

**Screenshots**

**Additional context**
https://github.com/rancher/dashboard/blob/3567111c26d19a974aa69919ec321081a198e1e9/shell/models/provisioning.cattle.io.cluster.js#L853-L855

### PR DESCRIPTION
**Pull request by momesgin**
### Summary
Fixes #14828 

### Occurred changes and/or fixed issues
This PR fixes a logical bug in the `supportsWindows` getter to correctly whitelist the Flannel CNI.

### Technical notes summary
- Updated `supportsWindows` logic in `shell/models/provisioning.cattle.io.cluster.js` to support `flannel`.
- Flannel has only supported Windows nodes since v1.29.2+rke2r1. This PR takes that into consideration but it's an edge case.
- Unit tests.

### Areas or cases that should be tested
- Verify that Windows support is correctly enabled/disabled based on the selected CNI (Calico or Flannel) for RKE2 clusters.
- Regarding the Flannel's supported Windows nodes since v1.29.2+rke2r1, a unit test has been added but it would be good if we could test that scenario as well.

### Areas which could experience regressions
- Checking all the CNI options to work as before should cover regressions.

### Screenshot/Video

image

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes
- [x] The PR has been reviewed in terms of Accessibility
- [x] The PR has considered, and if applicable tested with, the three Global Roles `Admin`, `Standard User` and `User Base`

### CODE REVIEW

File: shell/models/__tests__/provisioning.cattle.io.cluster.test.ts [hunk 1]
```diff
@@ -275,4 +275,86 @@ describe('class ProvCluster', () => {
...
+        },
+        expected: true
+      },
+      {
+        description: 'should return true if cni is flannel',
+        clusterData: {
+          isRke2: true, kubernetesVersion: 'v1.35.0', spec: { rkeConfig: { machineGlobalConfig: { cni: 'flannel' } } }
+        },
+        expected: true
+      },
```
[the remark this answers is not in the document]
Comment (momesgin) [reply shell/models/__tests__/provisioning.cattle.io.cluster.test.ts:350]: done

File: shell/models/provisioning.cattle.io.cluster.js [hunk 2]
```diff
@@ -897,7 +897,7 @@ export default class ProvCluster extends SteveModel {
 
     const cni = this.spec?.rkeConfig?.machineGlobalConfig?.cni;
```
[the remark this answers is not in the document]
Comment (momesgin) [reply shell/models/provisioning.cattle.io.cluster.js:899]: done
